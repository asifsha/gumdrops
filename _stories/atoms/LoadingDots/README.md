The `<LoadingDots>` component is used to indicate asynchronous loading of content.
