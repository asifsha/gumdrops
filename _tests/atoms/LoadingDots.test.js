/* globals mount */
import React from 'react';
import LoadingDots from '../../components/atoms/LoadingDots';

const defaultProps = {
    className: 'cool-class',
    size: 'lg',
    style: { width: '100px' }
};

describe('Expect <LoadingDots>', () => {
    it('to render', () => {
        const wrapper = mount(<LoadingDots {...defaultProps} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('to render as white', () => {
        const props = {
            ...defaultProps,
            isWhite: true
        };

        const wrapper = mount(<LoadingDots {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
