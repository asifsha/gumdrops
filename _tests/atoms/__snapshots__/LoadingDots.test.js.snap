// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <LoadingDots> to render 1`] = `
<LoadingDots
  className="cool-class"
  size="lg"
  style={
    Object {
      "width": "100px",
    }
  }
  whiteDots={false}
>
  <div
    className="cool-class"
    style={
      Object {
        "width": "100px",
      }
    }
  >
    <div
      className="gds-loading"
    >
      <div
        className="gds-loading__dot gds-loading__dot--lg"
      />
    </div>
  </div>
</LoadingDots>
`;

exports[`Expect <LoadingDots> to render as white 1`] = `
<LoadingDots
  className="cool-class"
  isWhite={true}
  size="lg"
  style={
    Object {
      "width": "100px",
    }
  }
  whiteDots={false}
>
  <div
    className="cool-class"
    style={
      Object {
        "width": "100px",
      }
    }
  >
    <div
      className="gds-loading"
    >
      <div
        className="gds-loading__dot gds-loading__dot--lg"
      />
    </div>
  </div>
</LoadingDots>
`;
